torchtext/experimental/transforms.py

```diff
@@ -176,6 +177,143 @@ def to_ivalue(self):
+            - text_bpe_25000
+            - text_bpe_50000
+            Otherwise, the file path to the user-provided sentencepiece model is required.
+
+    Examples:
+        >>> from torchtext.experimental.transforms import pretrained_spm
+        >>> sp_model = pretrained_spm('text_unigram_25000')
+
+    """
+    if spm_model in _pretrained_spm:
+        spm_model = download_from_url('https://pytorch.s3.amazonaws.com/models/text/pretrained_spm/{}.model'.format(spm_model))
+        return load_sp_model(spm_model)
+    else:
+        raise RuntimeError('The pretrained sentencepiece model is not valid')
+
+
+def sentencepiece_tokenizer(spm_model):
+    r"""factory function to generate SentencePieceTokenizer with pretrained sp model
+
+    Args:
+       spm_model: the sentencepiece model instance
+
+    Examples:
+        >>> import torch
+        >>> from torchtext.experimental.transforms import SentencePieceTokenizer
+        >>> from torchtext.experimental.transforms import pretrained_spm
+        >>> sp_model = pretrained_spm('text_unigram_25000')
+        >>> spm_tokenizer = sentencepiece_tokenizer(sp_model)
+        >>> jit_spm_tokenizer = torch.jit.script(spm_tokenizer)
+    """
+    return SentencePieceTokenizer(spm_model)
+
+
+class SentencePieceTokenizer(nn.Module):
+    r"""Tokenizer based on a pretained sentencepiece model.
+
+    Args:
+       spm_model: the sentencepiece model instance
+    """
+
+    def __init__(self, spm_model):
+        super(SentencePieceTokenizer, self).__init__()
+        self.sp_model = spm_model
+
+    def forward(self, line: str) -> List[str]:
+        r"""
+        Args:
+            line: the input sentence string
+
+        Examples:
+            >>> spm_tokenizer('the pretrained sp model names')
+            >>> ['▁the', '▁pre', 'trained', '▁sp', '▁model', '▁names']
+        """
+        return self.sp_model.EncodeAsPieces(line)
+
+    @torch.jit.export
+    def decode(self, tokens: List[str]) -> str:
+        r"""
+        Args:
+            tokens: the tokens list for decoder
+
+        Examples:
+            >>> spm_transform.decoder(['▁the', '▁pre', 'trained', '▁sp', '▁model', '▁names'])
+            >>> 'the pretrained sp model names'
+        """
+        return self.sp_model.DecodePieces(tokens)
+
+
+def sentencepiece_transform(spm_model):
+    r"""factory function to generate SentencePieceTransform with pretrained sp model
+
+    Args:
+       spm_model: the sentencepiece model instance
+
+    Examples:
+        >>> import torch
+        >>> from torchtext.experimental.transforms import SentencePieceTransform
+        >>> from torchtext.experimental.transforms import pretrained_spm
+        >>> sp_model = pretrained_spm('text_unigram_25000')
+        >>> spm_transform = sentencepiece_transform(sp_model)
+        >>> jit_spm_tokenizer = torch.jit.script(spm_transform)
+    """
+    return SentencePieceTransform(spm_model)
+
+
+class SentencePieceTransform(nn.Module):
```

Comment:
Should we call this a Vocab?

Reply:
Not really. It goes beyond Vocab as this transform combine Vocab/Tokenizer together.

Reply:
We could also add an nn.Module wrapper + forward method for the C++ SentencePiece class instead of introducing another class, that I think does what a user would expect SentencePiece to do by default.